### ISSUE TITLE
Quarkus MDC propagation still not working

### ISSUE THREAD

**Issue by AleDiggio** [quarkusio/quarkus#38452]
For a monitoring purpose I need to add contextual information to logs. I've seen the Quarkus documentation and seems that MCD is implemented but in my case the propagation doesn't work. I put the (key, value) pair in the mdc in a rest call and trying to access it in a Vertx executing block but the result is null.

**Comment by AleDiggio** [comment]
`   
import io.vertx.core.Vertx;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;
import org.jboss.logmanager.MDC;
import io.quarkus.logging.Log;



    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {

        MDC.put("request.id", UUID.randomUUID().toString());

        Log.info("GET received");
        this.testBlocking();

        return "Hello!!!";

    }

    public void testBlocking() {
        Vertx vertx = Vertx.vertx();
        vertx.executeBlocking(
                () -> {
                    Log.info("Blocking code !");

                    for(int i = 0; i < 10; i++) {
                        Log.info(i + " inside blocking code!");
                    }

                    return true;
                }
        ).onComplete(res -> Log.info("ssss"));
    }`

**Comment by brunobat** [comment]
@AleDiggio are you configuring the logger to print the MDC fields, like described here: https://quarkus.io/guides/logging#use-mdc-to-add-contextual-log-information

**Comment by AleDiggio** [comment]
@brunobat yes, i followed the guide word by word without adding third-part library or something like that. 

> In Quarkus, the MDC provider has a specific implementation for handling the reactive context, ensuring that MDC data is propagated during reactive and asynchronous processing.

**Comment by brunobat** [comment]
CC @cescoffier

**Comment by cescoffier** [comment]
Are you using RestEasy reactive?

**Comment by cescoffier** [comment]
Oh, you are creating a second vertx instance. Do NOT do that. Use @Inject Vertx vertx

**Comment by AleDiggio** [comment]
@cescoffier first of all thank you for your answer and your time, now it works! One last question: is the MDC propogation guarantee in every case?. For example in a situation like that: 

`    void onStart(@Observes StartupEvent ev) {
        MDC.put("request.id", UUID.randomUUID().toString());
    }`

Can i access the field setted at startup everywhere during execution flow? Like inside a @Incoming annotated method to handle message from a queue?

**Comment by cescoffier** [comment]
The MDC is propagated 'along the flow' of the processing. So your example is slightly odd as I don't see how an observer method is going to trigger the incoming (do you plan to use an emitter?).

To give an example:

1. You receive a http request
2. You call a remote http service using an async call
3. You process the response 

The MDC is propagated along the flow.
